### ISSUE TITLE
Option to run PretrainedTransformer in eval mode

### ISSUE THREAD

**Issue by mahnerak** [allenai/allennlp#4895]
While setting `train_parameters` to `False` very often we also may consider disabling dropout/batchnorm, in other words, to run the pretrained model in eval mode.
We've done a little modification to `PretrainedTransformerEmbedder` that allows providing whether the token embedder should be forced to `eval` mode during the training phase.

Do you this feature might be handy? Should I open a PR?

**Comment by AkshitaB** [comment]
@mahnerak It does make sense to disable dropout/batchnorm when using a pretrained model as a token embedder. Please feel free to open a PR!
